examples/sst2/input_pipeline.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""SST-2 input pipeline."""
+
+import collections
+from typing import Dict, List, Set, Text, Tuple
+import os
+
+from absl import logging
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+
+
+# pylint: disable=no-member
+
+
+# pylint: disable=abstract-method
+class OrderedCounter(collections.Counter, collections.OrderedDict):
+  """A counter that remembers the order of added keys."""
+  pass
+
+
+def get_tokens(datasets: List[tf.data.Dataset]) -> Set[Text]:
+  """Returns a set with all unique tokens in the given datasets."""
+  counter = OrderedCounter()
+  for ds in datasets:  # Data set already tokenized here.
+    for example in tfds.as_numpy(ds):
+      tokens = whitespace_tokenize(example['sentence'].strip())
+      counter.update(tokens)
+  sst_tokens = set(counter.keys())
+  logging.info('Number of unique tokens: %d', len(sst_tokens))
+  return sst_tokens
+
+
+def save_vocab(vocab, path):
+  """Saves a vocabulary to disk."""
+  with open(path, mode='wb') as f:
+    for token in vocab:
+      f.write(token + b'\n')
+  logging.info('Saved vocab to: %s', path)
+
+
+def load_vocab(path):
+  """Loads a vocabulary from disk."""
+  vocab = collections.OrderedDict()
+  with open(path, mode='rb') as f:
```

Comment:
I think 'r' is the default so you can omit.

```suggestion
  with open(path, mode='b') as f:
```

But feel free to keep it if you think it is clearer...

Reply:
Maybe I'll keep 'rb' if you're OK with it.